File: kairospy/infrastructure/integrations/domain/capabilities.py
from __future__ import annotations

from enum import StrEnum


class IntegrationCapability(StrEnum):
    MARKET_DATA = "market.data"
    MARKET_STREAM = "market.stream"
    REFERENCE_DATA = "reference.data"
    ACCOUNT_READ = "account.read"
    ACCOUNT_MARKET_PROFILE_READ = "account.market_profile.read"
    ACCOUNT_STREAM = "account.stream"
    ORDER_ENTRY = "order.entry"
    ORDER_CANCEL = "order.cancel"
    EXECUTION_STREAM = "execution.stream"
    EARN = "earn.operations"


__all__ = ["IntegrationCapability"]

File: crates/kairos-integration/src/domain/capabilities.rs
#[derive(Clone, Copy, Debug, Eq, PartialEq)]
pub enum IntegrationCapability {
    Reference,
    MarketData,
    AccountRead,
    AccountCredentialInspection,
    AccountMarketProfileRead,
    AccountStream,
    ExecutionStream,
    OrderEntry,
    OrderRead,
    MarketStream,
    Transfer,
    Earn,
}
